modules/compute/gke-node-pool/reservation_definitions.tf

```diff
@@ -66,4 +66,16 @@ locals {
   # Know that in map comparison the order of keys does not matter. That is {NVME: x, SCSI: y} and {SCSI: y, NVME: x} are equal
   # As of this writing, there is only one reservation supported by the Node Pool API. So, directly accessing it from the list
   specific_reservation_requirement_violations = length(local.reservation_vm_properties) == 0 ? [] : [for k, v in local.nodepool_vm_properties : k if v != local.reservation_vm_properties[0][k]]
+
+  specific_reservation_requirement_violation_messages = {
```

Comment:
In the absence of integration tests, can you please add to the PR results from a manual test which exercise these conditions ?

Reply:
Done